docs/design/features/byreflike-generics.md

```diff
@@ -0,0 +1,75 @@
+# Generics parameters of ByRefLike types
+
+Using ByRefLike types in Generic parameters is possible by building upon support added for `ref` fields. Scenarios that would benefit most from this are those involving `Span<T>`. For example, consider the following examples:
+
+- `Span<TypedReference>` &ndash; Represents the general case where a ByRefLike type is used as a Generic parameter. This specific case would be desirable for a more efficient Reflection API.
+- `Span<Span<char>>` &ndash; Nested `Span<T>` types would be of benefit in the parsing result of strings.
+
+## Runtime impact
+
+Supporting ByRefLike type as Generic parameters will impact the following IL instructions:
+
+- `box` &ndash; Types with ByRefLike parameters used in fields cannot be boxed.
+- `throw` &ndash; Requires an object reference on the stack so not directly impacted since boxing is not permitted.
+- `stsfld` / `ldsfld` &ndash; Type fields of a ByRefLike parameter cannot be marked `static`.
+- `newarr` / `stelem` / `ldelem` / `ldelema` &ndash; Arrays are not able to contain ByRefLike types.
+    - `newobj` &ndash; For multi-dimensional array construction.
+- `constrained.callvirt` &ndash; This IL sequence must resolve to a method implemented on `object`, or a default interface method.
+- Use of any Generic which doesn’t expect a ByRefLike parameter as a Generic parameter.
```

Comment:
Should `isinst` and `unbox.any` be included?

Reply:
I don't think we can get to the `unbox.any` instruction since we can't box. Not saying I shouldn't call out behavior, but do want to make sure I'm understanding the why it should be included.

For `isinst`, that is a good point. I _think_ you're right. Let me page back in how this instruction works so I can define the "why".

Reply:
For `unbox.any`, is the following possible?
```
.method static !!T F<T>(object obj)
{
    ldarg.0
    unbox.any !!T
    ret
}
```

Reply:
I think this will simply always fail because `obj` will never be able to be a `T` that is ByRefLike.